### PR TITLE
feat(implement-new-functionality-links): add label-in-name

### PR DESCRIPTION
**Pull request by shanisebarona**
#### Details

This PR adds `Label in name`  requirement to Assessments under the Links tab. 

##### Motivation

Feature work 🎇 

##### Context

The rule for `label-in-name` will be added in a separate PR, right now it's using the `link-purpose` rule as a placeholder.

#### Label in name page under Assessment > Links
image

#### Label in name info and examples guidance page
image
image

#### Pull request checklist

- [x] Addresses an existing issue: #3736
- [x] Ran `yarn null:autoadd`
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Review by madalynrose — APPROVED** [review]
This looks great! It might be good to either note the places we'll need to update once the custom rule is in or wait until that PR is merged to finish this one.

We should get sign off from @nang4ally that the copy looks good but the code gets approval from me!

**Comment by shanisebarona** [comment]
> This looks great! It might be good to either note the places we'll need to update once the custom rule is in or wait until that PR is merged to finish this one.
> 
> We should get sign off from @nang4ally that the copy looks good but the code gets approval from me!

Agreed! I will add in a comment where the custom rule will need to be updated.

**Review by nang4ally — APPROVED** [review]
Looks good to me!! :)

**Pull request merged**
